cni/network/invoker_azure.go

```diff
@@ -102,6 +102,10 @@ func (invoker *AzureIPAMInvoker) deleteIpamState() {
 		return
 	}
 
+	if cniStateExists {
+		return
```

Comment:
I think we should put a comment here explaining why we return if cniStateExists. Even for folks that have seen this code path before, it's not immediately obvious.